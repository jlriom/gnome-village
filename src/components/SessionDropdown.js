import React, { useState } from "react";
import history from "../shared/history";
import { NavDropdown } from "react-bootstrap";
import LogoutPage from "../containers/auth/LogoutPage";
import MyProfilePage from "../containers/user/MyProfilePage";
import UnRegisterPage from "../containers/user/UnRegisterPage";

const SessionDropdown = ({ auth }) => {
	const [showMyProfilePage, setShowMyProfilePage] = useState(false);
	const [showLogoutPage, setShowLogoutPage] = useState(false);
	const [showUnregisterPage, setShowUnregisterPage] = useState(false);

	return (
		<>
			<MyProfilePage
				show={showMyProfilePage}
				handleClose={() => setShowMyProfilePage(false)}></MyProfilePage>
			<LogoutPage
				show={showLogoutPage}
				handleClose={() => setShowLogoutPage(false)}></LogoutPage>

			<UnRegisterPage
				show={showUnregisterPage}
				handleClose={() => setShowUnregisterPage(false)}></UnRegisterPage>

			<NavDropdown title={auth.username} id='basic-nav-dropdown'>
				{!auth.isGuest && (
					<NavDropdown.Item onClick={() => setShowMyProfilePage(true)}>
						My profile
					</NavDropdown.Item>
				)}

				{!auth.isGuest && (
					<NavDropdown.Item
						onClick={() => history.push(`/brastlewark/${auth.id}`)}>
						My Friends
					</NavDropdown.Item>
				)}

				{!auth.isGuest && (
					<NavDropdown.Item onClick={() => setShowUnregisterPage(true)}>
						Unregister my profile
					</NavDropdown.Item>
				)}
				{!auth.isGuest && <NavDropdown.Divider />}
				<NavDropdown.Item onClick={() => setShowLogoutPage(true)}>
					Logout
				</NavDropdown.Item>
			</NavDropdown>
		</>
	);
};

export default SessionDropdown;
