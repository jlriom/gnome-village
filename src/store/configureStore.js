import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
