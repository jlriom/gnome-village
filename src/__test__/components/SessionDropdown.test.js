import React from "react";
import SessionDropdown from "../../components/SessionDropdown";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import configureStore from "../../store/configureStore";
const store = configureStore();

const auth = {
    username: "username",
    isGuest: false
}

describe("<SessionDropdown />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <SessionDropdown auth={auth} />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
