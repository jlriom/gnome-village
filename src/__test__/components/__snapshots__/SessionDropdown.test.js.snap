// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionDropdown /> renders ok 1`] = `
<div
  className="dropdown nav-item"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle nav-link"
    href="#"
    id="basic-nav-dropdown"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    username
  </a>
</div>
`;
